convex/lib/validators.ts

```diff
@@ -0,0 +1,37 @@
+import { Validator, v } from 'convex/values'
+import { Doc, TableNames } from '../_generated/dataModel'
+import schema from '../schema'
+
+export const betterV = {
+  id: <T extends TableNames>(tableName: T) => {
+    return v.id(tableName)
+  },
+  doc: <T extends TableNames>(tableName: T): Validator<Doc<T>, false, any> => {
+    const validator = v.object(
+      (schema.tables[tableName] as any).documentSchema
```

Comment:
I made normal Convex schema expose the validator as `documentType`. Ents have a private `documentSchema`, but I think we should ideally make it behave like `documentType` with vanilla Convex